lms/resources/_js_config.py

```diff
@@ -60,6 +63,46 @@ def asdict(self):
         """
         return self._config
 
+    def enable_oauth_error_mode(
+        self, error_details, is_scope_invalid=False, requested_scopes=None
+    ):
+        """
+        Put the JavaScript code into "OAuth authorization failed" mode.
+
+        This mini-app is shown when OAuth authorization with Canvas fails
+        after redirecting to Canvas's OAuth authorization endpoint.
+
+        :param error_details: Technical details of the error
+        :type error_details: str
+        :param is_scope_invalid: `True` if authorization failed because the
+          OAuth client does not have access to all the necessary scopes
+        :type is_scope_invalid: bool
+        :param requested_scopes: List of Canvas API scopes that were requested
+        :type requested_scopes: list(str)
+        """
+        self._config.update(
+            {
+                "mode": "canvas-auth-error-dialog",
+                "invalidScope": is_scope_invalid,
+                "errorDetails": error_details,
+                "scopes": requested_scopes or [],
```

Comment:
```suggestion
                "oauth2_redirect_error": {
                    "invalidScope": is_scope_invalid,
                    "errorDetails": error_details,
                    "scopes": requested_scopes or [],
                },
```

Can we move these three settings into an `"oauth2_redirect_error"` sub-object? We've been trying to tidy up the config by grouping things into sub-objects